File: javamodule/src/main/java/com/timmy/javamodule/MainActivity.java
package com.timmy.javamodule;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.timmy.javamodule.viewmodel.ViewModelStuActivity;

/**
 * JetPacket java语言实现
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getLifecycle().addObserver(new MyObserver());
    }

    @Override
    protected void onResume() {
        super.onResume();
        TLog.d("onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        TLog.d("onPause");
    }

    public void jump(View view) {
        startActivity(new Intent(this, LiveDataStuActivity.class));
    }

    public void jumpMVVM(View view) {
        startActivity(new Intent(this, ViewModelStuActivity.class));
    }

    private class MyObserver implements LifecycleObserver {
        @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
        void onResume() {
            TLog.d("MyObserver -- onPause");
        }

        @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
        void onPause() {
            TLog.d("MyObserver -- onPause");
        }
    }
}

File: app/src/main/java/com/timmy/kotlin/ui/MainActivity.kt
package com.timmy.kotlin.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.lifecycle.LifecycleObserver
import com.timmy.kotlin.App
import com.timmy.kotlin.R
import kotlinx.android.synthetic.main.activity_main.*

/**
 * 4个底部Tab
 * - Home，Hoke，框架，我的
 * 网络，H5页面
 *
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val applicationInfo = App.app.applicationInfo
        Log.e("Tim", applicationInfo.processName)

        var tv = findViewById<TextView>(R.id.tv_main)
        tv.setOnClickListener(View.OnClickListener {
            Toast.makeText(this, "Toast stet", Toast.LENGTH_SHORT).show()
        })

        tv_main?.run {
            this.setOnClickListener(View.OnClickListener {

            })
        }

        lifecycle.addObserver(MyObserver())

    }

    public class MyObserver : LifecycleObserver {

    }

}